programs/magicblock/src/schedule_task/process_schedule_task.rs

```diff
@@ -99,7 +104,13 @@ pub(crate) fn process_schedule_task(
                 return Err(InstructionError::MissingRequiredSignature);
             }
 
-            if !ix_accounts.contains(&account.pubkey) {
+            if let Some((_pk, writable)) =
+                ix_accounts.iter().find(|(pk, _)| *pk == &account.pubkey)
+            {
+                if *writable != account.is_writable {
```

Comment:
Actually this won't be needed after we decided to not check anything at the time of scheduling [see this comment](https://github.com/magicblock-labs/magicblock-validator/pull/493#discussion_r2324676088).
Sorry for the back and forth on this.

We may not even require the account to even exist either, assuming it'll be created before the task runs the first time.